src/cmd/run.go

```diff
@@ -205,6 +205,21 @@ func runCommand(container string,
 			fmt.Fprintf(os.Stderr, "Entering container %s instead.\n", container)
 			fmt.Fprintf(os.Stderr, "Use the 'create' command to create a different toolbox.\n")
 			fmt.Fprintf(os.Stderr, "Run '%s --help' for usage.\n", executableBase)
+		} else if defaultContainer && rootFlags.enterShortcut {
+			fmt.Fprintf(os.Stderr, "Default container does not exist. Creating it.\n")
```

Comment:
Umm... we shouldn't add a separate criterion for the `toolbox` with no commands and flags case. We should stick to the existing set of criteria that's used to make `toolbox enter` a bit more flexible.

Our objective isn't to have exact message by message compatibility with `coreos/toolbox`. That's fundamentally impossible because Toolbox has a lot many more features and users using it across very different environments, whereas `coreos/toolbox` is a relatively small shell script that's used in a very specific way. eg., we specifically don't want entering just `toolbox` to potentially start downloading 100s of MBs, because that will be very surprising to those who have just stumbled across `toolbox(1)`.

I think the idea is to have some rough compatibility, where the two things aren't completely different.

In this case, I'd expect `toolbox` to behave like `toolbox enter`. For absolute beginners, who might have read about `coreos/toolbox`, Toolbox would still behave in somewhat the same way, even though it might prompt them for various things. As long as the prompts are self-explaining, that's fine. One can also imagine `coreos/toolbox` growing such prompts in the future.

Compatibility or not, ultimately, we need to ensure that the tool continues to behave in a coherent, understandable and easy to document way. The more special cases we add, the less coherent and understandable it becomes, and the harder it gets to document it.